deployment/secure_api_server.py

```
@@ -87,95 +87,122 @@
         except Exception as e:
             # Release rate limit slot on error
             rate_limiter.release_request(client_ip, user_agent)
-            
+
             response_time = time.time() - start_time
-            update_metrics(response_time, success=False, error_type='endpoint_error')
-            logger.error(f"Endpoint error: {str(e)}")
-            return jsonify({'error': str(e)}), 500
-    
+            update_metrics(response_time, success=False, error_type="endpoint_error")
+            logger.error(f"Endpoint error: {e!s}")
```

Ruff (0.13.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)